validator/remote/src/main/java/tech/pegasys/teku/validator/remote/BeaconNodeReadinessManager.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright ConsenSys Software Inc., 2022
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.validator.remote;
+
+import com.google.common.collect.Maps;
+import java.time.Duration;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Stream;
+import okhttp3.HttpUrl;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import tech.pegasys.teku.infrastructure.async.AsyncRunner;
+import tech.pegasys.teku.infrastructure.async.Cancellable;
+import tech.pegasys.teku.infrastructure.async.SafeFuture;
+import tech.pegasys.teku.service.serviceutils.Service;
+
+public class BeaconNodeReadinessManager extends Service {
+
+  private static final Logger LOG = LogManager.getLogger();
+
+  private final Map<RemoteValidatorApiChannel, Boolean> readinessStatusCache =
+      Maps.newConcurrentMap();
+
+  private final RemoteValidatorApiChannel primaryBeaconNodeApi;
+  private final List<RemoteValidatorApiChannel> failoverBeaconNodeApis;
+  private final AsyncRunner asyncRunner;
+  private final RemoteBeaconNodeSyncingChannel remoteBeaconNodeSyncingChannel;
+  private final Duration beaconNodeSyncingQueryPeriod;
+
+  private volatile Cancellable syncingQueryTask;
+
+  public BeaconNodeReadinessManager(
+      final RemoteValidatorApiChannel primaryBeaconNodeApi,
+      final List<RemoteValidatorApiChannel> failoverBeaconNodeApis,
+      final AsyncRunner asyncRunner,
+      final RemoteBeaconNodeSyncingChannel remoteBeaconNodeSyncingChannel,
+      final Duration beaconNodeSyncingQueryPeriod) {
+    this.primaryBeaconNodeApi = primaryBeaconNodeApi;
+    this.failoverBeaconNodeApis = failoverBeaconNodeApis;
+    this.asyncRunner = asyncRunner;
+    this.remoteBeaconNodeSyncingChannel = remoteBeaconNodeSyncingChannel;
+    this.beaconNodeSyncingQueryPeriod = beaconNodeSyncingQueryPeriod;
+  }
+
+  public boolean isReady(final RemoteValidatorApiChannel beaconNodeApi) {
+    return readinessStatusCache.getOrDefault(beaconNodeApi, true);
+  }
+
+  @Override
+  protected SafeFuture<?> doStart() {
+    syncingQueryTask =
+        asyncRunner.runWithFixedDelay(
+            () -> {
+              final SafeFuture<Void> primaryReadinessCheck = performPrimaryReadinessCheck();
+              final Stream<SafeFuture<?>> failoverReadinessChecks =
+                  failoverBeaconNodeApis.stream().map(this::performFailoverReadinessCheck);
```

Comment:
this could be a terrible idea... just prefacing with that...
really since we're testing all nodes all the time, we could just have a single list, where item 0 is primary and everything else is failover...
then theres no real difference between primary / non primary, although i guess theres no point flapping between failovers, you'd potentially still be looking if element 0 is available where you're checking primary... anyway just a thought, then the stream here checks all nodes...

Reply:
Yeah, I get your point. I am mainly having the primary separate to make use of the `RemoteBeaconNodeSyncingChannel`  primary node not in sync/ back in sync callbacks.